### PR TITLE
Migrate dwds/web directory to null safety

### PR DESCRIPTION
**Pull request by annagrin**
Migrate dwds/web directory to null safety

### CODE REVIEW

File: dwds/web/promise.dart [hunk 1]
```diff
@@ -49,7 +47,7 @@ Future<T> toFuture<T>(Promise<T> promise) {
   final completer = Completer<T>();
   promise.then(
     allowInterop(completer.complete),
-    allowInterop(completer.completeError),
+    allowInterop((e) => completer.completeError(e)),
```
Comment (elliette) [inline dwds/web/promise.dart:50]: Why is this necessary?

File: dwds/web/reloader/legacy_restarter.dart [hunk 2]
```diff
@@ -22,16 +20,17 @@ class LegacyRestarter implements Restarter {
...
       if (message is Map &&
           message['type'] == 'DDC_STATE_CHANGE' &&
           message['state'] == 'restart_end') {
         reloadCompleter.complete(true);
-        sub.cancel();
       }
     });
-    return reloadCompleter.future;
+    return reloadCompleter.future.then((value) {
+      sub.cancel();
```
Comment (elliette) [inline dwds/web/reloader/legacy_restarter.dart:32]: Why was this moved into the `then`?

File: dwds/web/reloader/require_restarter.dart [hunk 3]
```diff
@@ -54,7 +53,7 @@ class RequireLoader {
   external String get digestsPath;
 
   @JS()
-  external JsMap<String, List<String>> get moduleParentsGraph;
+  external JsMap<String, List<String>?> get moduleParentsGraph;
```
Comment (elliette) [inline dwds/web/reloader/require_restarter.dart:56]: Does it make sense for some of the values to be `null`? Maybe we could do a null check before adding them to the map?

File: dwds/web/promise.dart (hunk 1 above)
Comment (annagrin) [reply dwds/web/promise.dart:50]: it does not compile due to completer.completeError having an optional stackTrace parameter.

File: dwds/web/reloader/legacy_restarter.dart (hunk 2 above)
Comment (annagrin) [reply dwds/web/reloader/legacy_restarter.dart:32]: the sub is not defined inside the callback to `window.onMessage.listen` yet (at least analyzer does not think so:) - so i moved it to where it logically behaves the same but cancels the sub as before.

File: dwds/web/promise.dart (hunk 1 above)
Comment (elliette) [reply dwds/web/promise.dart:50]: Got it, thanks!

File: dwds/web/reloader/require_restarter.dart (hunk 3 above)
Comment (annagrin) [reply dwds/web/reloader/require_restarter.dart:56]: Yes, good point! Looks like they already are non-null, changed.

**Review by elliette — APPROVED** [review]
